src/main/java/one/util/streamex/AbstractStreamEx.java

```diff
@@ -252,6 +254,61 @@ public S skip(long n) {
         return supply(stream().skip(n));
     }
 
+    /**
+     * Returns a new, non-splittable, sequential stream containing
+     * all the elements of the original stream except the last n elements.
+     * Consumes space proportional to the number of elements skipped.
+     *
+     * <p>
+     * For example, {@code StreamEx.of("a", "b", "c").skipLast(1)} will yield a stream containing
+     * two elements: a, b.
+     *
+     * <p>
+     * This is an <a href="package-summary.html#StreamOps">quasi-intermediate operation</a>.
+     *
+     * @param n the non-negative number of elements to leave off the end of the stream.
+     * @return the new, sequential stream
+     * @throws IllegalArgumentException if {@code n} is negative or {@link Integer#MAX_VALUE}
+     * @since 0.6.7
+     */
+    public S skipLast(int n) {
+        if (n < 0 || n == Integer.MAX_VALUE)
+            throw new IllegalArgumentException(Long.toString(n));
+        if (n == 0)
+            return supply(this);
+
+        BlockingDeque<T> buffer = new LinkedBlockingDeque<>(n + 1);
+        Spliterator<T> source = this.spliterator();
+
+        return supply(StreamEx.of(new Spliterator<T>() {
+
+            @Override
+            public boolean tryAdvance(Consumer<? super T> action) {
+                while (buffer.remainingCapacity() > 0 && source.tryAdvance(buffer::offer)) { }
+                if (buffer.size() > n) {
+                    action.accept(buffer.poll());
+                    return true;
+                }
+                return false;
+            }
+
+            @Override
+            public Spliterator<T> trySplit() {
+                return null;
+            }
+
+            @Override
+            public long estimateSize() {
+                return source.estimateSize() - n;
+            }
+
+            @Override
+            public int characteristics() {
+                return source.characteristics();
+            }
+        })).sequential();
```

Comment:
A `forEachRemaining` method must be provided as it optimizes the common case (non-short-circuiting stream). In specific cases it could be much faster and memory friendly than `tryAdvance`.